gapic/src/main/com/google/gapid/perfetto/views/State.java

```diff
@@ -255,6 +254,10 @@ public void setSelection(ListenableFuture<Selection.MultiSelection> futureSel) {
   public void setSelection(Selection.MultiSelection selection) {
     lastSelectionUpdateId.incrementAndGet();
     this.selection = selection;
+    // If selection is cleared or set to a non-cpu one, don't do color grouping for cpu slices.
+    if (selection == null || selection.getSelection(Selection.Kind.Cpu) == Selection.EMPTY_SELECTION) {
```

Comment:
Instead of comparing against the empty selection, we should add a `isEmpty()` method.

Reply:
Done in the new commit.